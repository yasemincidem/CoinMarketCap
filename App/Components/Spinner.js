// @flow
import React from 'react';
import ReactNativeSpinner from 'react-native-spinkit';
import * as colors from '../Theme/Colors'

type Props = {
  color: string,
};

const Spinner = (props: Props) => <ReactNativeSpinner type="Wave" color={props.color} />;

Spinner.defaultProps = {
  color: colors.primary,
};

export default Spinner;
