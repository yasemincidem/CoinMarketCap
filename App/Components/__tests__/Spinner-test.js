import React from 'react';
import {shallow} from "react-native/jest/renderer";
import Spinner from '../Spinner';

describe('Spinner', () => {
  const defaultProps = {};

  const createWrapper = (overriddenProps) => {
    const props = {
      ...defaultProps,
      ...overriddenProps,
    };

    return shallow(<Spinner {...props} />);
  };

  it('renders correctly', () => {
    const wrapper = createWrapper();

    expect(wrapper).toMatchSnapshot();
  });
});
