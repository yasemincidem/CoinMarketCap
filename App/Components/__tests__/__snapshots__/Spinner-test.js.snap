// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders correctly 1`] = `
<Spinner
  color="#005DE8"
/>
`;
